packages/api/src/utils/client/request.ts

```diff
@@ -102,7 +102,7 @@ export function createApiRequest<E extends Endpoint>(
   return new Request(url, {
     ...init,
     method: definition.method,
-    headers: mergeHeaders(headers, req.headers),
+    headers: mergeHeaders(headers, req.headers, init.headers),
```

Comment:
This is anyways better imo, we should allow the users to override any header manually